### PR TITLE
HPCC-26766 Improve Prometheus Sink logging and response status

### PR DESCRIPTION
**Pull request by rpastrana**
- Adds minimal req and resp logging entries
- Improves TxSummary
- Sets successful response HTTP code

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by rpastrana** [comment]
@wangkx

**Comment by rpastrana** [comment]
@AttilaVamos the core file doesn't seem related, is this a known issue?
Can we retry the smoketest?

[New LWP 13664]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
Core was generated by `/var/lib/HPCCSystems/mythor/thormaster_mythor --master=10.20.0.91:20000'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007f88c44562dd in pthread_getschedparam () from /lib64/libpthread.so.0
[Current thread is 1 (Thread 0x7f88cc696140 (LWP 13648))]

**Review by wangkx — COMMENTED** [review]
@rpastrana a few minor comments

File: system/metrics/sinks/prometheus/prometheusSink.cpp [hunk 1]
```diff
@@ -64,16 +64,21 @@ PrometheusMetricSink::PrometheusMetricSink(const char *name, const IPropertyTree
...
             res.set_content(respmessage.str(), PROMETHEUS_METRICS_SERVICE_RESP_TYPE);
 
-            LOG(MCuserInfo, "TxSummary[status=%d;user=@%s:%d;contLen=%ld;req=%s;]\n", res.status, req.remote_addr.c_str(), req.remote_port, req.content_length, req.method.c_str());
+            LOG(MCuserError, "PrometheusMetricsService Error: %s",msg.str());
+            LOG(MCuserInfo, "TxSummary[status=%d;user=@%s:%d;contLen=%ld;req=%s;path=%s]", res.status, req.remote_addr.c_str(), req.remote_port, req.content_length, req.method.c_str(), req.path.c_str());
         });
 
         m_server.Get(m_metricsServiceName.str(), [&](const Request& req, Response& res)
         {
+            LOG(MCdebugInfo, "GET PrometheusMetricsService%s, from %s:%d ",req.path.c_str(), req.remote_addr.c_str(), req.remote_port);
```
[the remark this answers is not in the document]
Comment (rpastrana) [reply system/metrics/sinks/prometheus/prometheusSink.cpp:73]: intentional. It'll show up as "GET PrometheusMetricsService/metrics

**Comment by rpastrana** [comment]
@wangkx minor format issues addressed. Please review.

**Review by wangkx — APPROVED** [review]
@rpastrana looks fine.

**Comment by rpastrana** [comment]
@richardkchapman ready for merge

**Pull request merged**